extensions/objects/templates/FilepathEditor.js

```diff
@@ -31,109 +31,118 @@ oppia.directive('filepathEditor', [
       scope: true,
       template: '<div ng-include="getTemplateUrl()"></div>',
       controller: ['$scope', function($scope) {
-        // Reset the component each time the value changes (e.g. if this is part
-        // of an editable list).
+        // Reset the component each time the value changes
+        // (e.g. if this is part of an editable list).
         $scope.$watch('$parent.value', function(newValue) {
-          $scope.localValue = {
-            label: newValue || ''
-          };
-          $scope.imageUploaderIsActive = false;
+          if (newValue) {
+            $scope.setSavedImageUrl(newValue, false);
+          }
         });
 
-        $scope.explorationId = explorationContextService.getExplorationId();
+        $scope.validate = function(state) {
+          return state.state === 'saved' &&
+                 state.savedUrl &&
+                 state.savedUrl.length > 0;
+        };
 
-        $scope.validate = function(localValue) {
-          return localValue.label && localValue.label.length > 0;
+        $scope.isNoImageUploaded = function() {
+          return !$scope.state || $scope.state.state === 'empty';
         };
 
-        $scope.$watch('localValue.label', function(newValue) {
-          if (newValue) {
-            alertsService.clearWarnings();
-            $scope.localValue = {
-              label: newValue
-            };
-            $scope.$parent.value = newValue;
-          }
-        });
+        $scope.isImageUploaded = function() {
+          return $scope.state && $scope.state.state === 'uploaded';
+        };
 
-        $scope.getPreviewUrl = function(filepath) {
-          var encodedFilepath = window.encodeURIComponent(filepath);
-          return $sce.trustAsResourceUrl(
-            '/imagehandler/' + $scope.explorationId + '/' + encodedFilepath);
+        $scope.isImageSaved = function() {
+          return $scope.state && $scope.state.state === 'saved';
         };
 
-        $scope.resetImageUploader = function() {
-          $scope.currentFile = null;
-          $scope.currentFilename = null;
-          $scope.imagePreview = null;
+        $scope.setEmptyState = function() {
```

Comment:
Well to be consistent with the changes above, I've called this clearScopeData. Thoughts?